x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/Validations.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression;
+
+import org.elasticsearch.xpack.ql.common.Failure;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Expression.TypeResolution;
+import org.elasticsearch.xpack.ql.expression.TypeResolutions;
+
+public final class Validations {
+
+    private Validations() {}
+
+    public static Failure isLiteral(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
```

Comment:
I think `isLiteral` may give the wrong impression.
```suggestion
    public static Failure isFoldable(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
```